docker/1.13/Dockerfile.cpu

```diff
@@ -1,33 +1,77 @@
-ARG TFS_VERSION
-
-FROM tensorflow/serving:${TFS_VERSION} as tfs
 FROM ubuntu:18.04
+
+LABEL maintainer="Amazon AI"
 LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
+
+ARG PYTHON=python3
+ARG PIP=pip3
+ARG TFS_SHORT_VERSION=1.13
 
-COPY --from=tfs /usr/bin/tensorflow_model_server /usr/bin/tensorflow_model_server
+# See http://bugs.python.org/issue19846
+ENV LANG C.UTF-8
+# Python won’t try to write .pyc or .pyo files on the import of source modules
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV SAGEMAKER_TFS_VERSION="${TFS_SHORT_VERSION}"
+ENV PATH="$PATH:/sagemaker"
+ENV LD_LIBRARY_PATH='/usr/local/lib:$LD_LIBRARY_PATH'
+ENV MODEL_BASE_PATH=/models
+# The only required piece is the model name in order to differentiate endpoints
+ENV MODEL_NAME=model
 
 # nginx + njs
-RUN \
-    apt-get update && \
-    apt-get -y install --no-install-recommends curl gnupg2 && \
-    curl -s http://nginx.org/keys/nginx_signing.key | apt-key add - && \
-    # find the correct linux distribution here - http://nginx.org/en/linux_packages.html
-    echo 'deb http://nginx.org/packages/ubuntu/ bionic nginx' >> /etc/apt/sources.list && \
-    apt-get update && \
-    apt-get -y install --no-install-recommends nginx nginx-module-njs python3 python3-pip python3-setuptools && \
-    apt-get clean
-
-# cython, falcon, gunicorn, tensorflow-serving
-RUN \
-    pip3 install --no-cache-dir cython falcon gunicorn gevent requests grpcio protobuf && \
-    pip3 install --no-dependencies --no-cache-dir tensorflow-serving-api==1.13.0
+RUN apt-get update \
+ && apt-get -y install --no-install-recommends curl gnupg2 ca-certificates git wget vim build-essential zlib1g-dev \
+ && curl -s http://nginx.org/keys/nginx_signing.key | apt-key add - \
+ && echo 'deb http://nginx.org/packages/ubuntu/ bionic nginx' >> /etc/apt/sources.list \
+ && apt-get update \
+ && apt-get -y install --no-install-recommends nginx nginx-module-njs python3 python3-pip python3-setuptools \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN ${PIP} --no-cache-dir install --upgrade pip setuptools
+
+# cython, falcon, gunicorn, grpc
+RUN ${PIP} install -U --no-cache-dir \
+    awscli==1.16.130 \
+    cython==0.29.10 \
+    falcon==2.0.0 \
+    gunicorn==19.9.0 \
+    gevent==1.4.0 \
+    requests==2.21.0 \
+    grpcio==1.24.1 \
+    protobuf==3.10.0
+
+# tensorflow-serving-api
+RUN ${PIP} install --no-dependencies --no-cache-dir \
```

Comment:
can we combine there pip installations with an &&

also can you add a comment for the --no-deps parameter, is it to avoid installing tensorflow?

Reply:
Yes, it is to avoid installing tensorflow.

updated.